### DISCUSSION TITLE
Integrating Solax and Octopus

### DISCUSSION THREAD

**Discussion by howard-paster** [wills106/homeassistant-solax-modbus#1064] [Q&A]
I have Octopus with an Intelligent Octopus for EVs tariff. Its really simple "Intelligent Dispatching" is set to on when I get super cheap electricity for EV charging and off when standard rate.
This is usually between 23:30 and 05:30 daily but can be extra slots during the day depending on if I need to charge car and/or excess energy on grid. 
I have Solax inverter with 20kw of batteries and PV array.
I need to be able to set inverter to charge batteries (house load draw from grid) when "Intelligent Dispatching" is set to true and when it switches back draw house load form batteries (and if excess PV send to grid). 

Reading through notes looks like I need some sort of automation where I set Enabled Battery control to something +ve to charge and zero to draw from Battery. Would like to utilise Modbus Power Control as seems a safe option but struggling to work out which one of the several example fits the bill as there doesn't seem to be such an exampl e- dont need the coimplexity of discharging battery to grid 

Has anyone done something like this - and have an example they can share

**Comment by gerd091986** [comment]
Intellegent Dispatching wasnt reliable for me for some reason, so I use when my rate changes:

WHEN:

Current Rate Electricity (XXXX) money changes - Below 0.08 as my cheap rate is 0.075

THEN DO:

Press SolaX Inverter Battery Awaken button
wait 1min
Change SolaX Inverter Charger Use Mode option - Manual Mode
wait 1min
 Change SolaX Inverter Manual Mode Select option - Force Charge
 
Then another automation for when the price goes up:
 
 WHEN:

 Current Rate Electricity (XXXX) money changes- above 0.08 as my cheap rate is 0.075

THEN DO:

Press SolaX Inverter Battery Awaken button
wait 1min
Change SolaX Inverter Charger Use Mode option - Self Use Mode

**Comment by howard-paster** [comment]
I was under the impression that Intelligent Dispatching  is on when you are on cheap rate either between 23:30 and 05:30 of if app decides to give cheap charging outside this time. The idea is to have an automation that wifi charge batteries if either is true. The logic is sound but not convinced that the logic works with the data. Charge when batteries <90% and either cheap rate electricity or dispatching is true  and car charging. Just tried by so a smart charge at 11am to top up battery, as outside normal window would expect Diaptaching to be true (but it wasn't) - is there another way of working at that outside of normal 23:30-05:30 window
 but still cheap rate

**Reply by gerd091986** [reply]
Yes, thats why I changes from "Intellegent Dispatching" to the "urrent Rate Electricity (XXXX) money changes"

It's works and has been very reliable for me for some time. 

Be careful if your battery is under 10% and IDLE it wont kick in, you need to keep the inverter awake.

**Comment by howard-paster** [comment]
I have tried twice last week current rate electricity dropped to 7p whilst charging - this time nothing. Last time dispatching changed not this time - maybe because only 5 min duration. will try again tomorrow
